File: frontend/src/services/TrainingService.ts
import { api }
from "./api";

import type { Training }
from "../types/training";

interface CreateTrainingDTO {

  category: string;

  duration: number;

  bpm: number;

  notes: string;

  instrumentId: string;
}

export async function getTrainings() {

  const response =
    await api.get<Training[]>("/training");

  return response.data;
}

export async function createTraining(
  data: CreateTrainingDTO
) {

  const response =
    await api.post(
      "/training",
      data
    );

  return response.data;
}

export async function deleteTraining(
  id: string
) {

  await api.delete(
    `/training/${id}`
  );
}

File: backend/music-study-manager/music-study-manager/src/main/java/com/moises/musicstudy/music_study_manager/training/service/TrainingService.java
package com.moises.musicstudy.music_study_manager.training.service;

import com.moises.musicstudy.music_study_manager.training.dto.CreateTrainingDTO;
import com.moises.musicstudy.music_study_manager.training.dto.TrainingResponseDTO;
import com.moises.musicstudy.music_study_manager.training.entity.TrainingSession;
import com.moises.musicstudy.music_study_manager.training.repository.TrainingRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;

@Service
@RequiredArgsConstructor
public class TrainingService {

    private final TrainingRepository repository;

    public TrainingResponseDTO create(CreateTrainingDTO dto) {

        TrainingSession session = TrainingSession.builder()
                .instrumentId(dto.getInstrumentId())
                .duration(dto.getDuration())
                .category(dto.getCategory())
                .bpm(dto.getBpm())
                .notes(dto.getNotes())
                .attachments(dto.getAttachments())
                .createdAt(LocalDateTime.now())
                .build();

        repository.save(session);

        return mapToDTO(session);
    }

    public List<TrainingResponseDTO> findAll() {

        return repository.findAll()
                .stream()
                .map(this::mapToDTO)
                .toList();
    }

    private TrainingResponseDTO mapToDTO(
            TrainingSession session) {

        return new TrainingResponseDTO(
                session.getId(),
                session.getDuration(),
                session.getCategory(),
                session.getBpm()
        );
    }

    public void delete(String id) {

        repository.deleteById(id);
    }
}